Add docs for publishing to RubyGems

alex-ju opened the pull request:
Add documentation for publishing to RubyGems.

[See rendered version](https://github.com/alphagov/govuk_publishing_components/blob/add-publishing-docs/docs/publishing-to-rubygems.md)

vanitabarrett commented on docs/publishing-to-rubygems.md:
```diff
@@ -0,0 +1,29 @@
+
+1. Checkout **master** and pull latest changes.
+
+2. Create and checkout a new branch (`release-[version-number]`).
```
Is it worth mentioning to check the latest commits to make sure they're included in the Changelog? People unfamiliar with working in the repo can forget to add to the Changelog as Unreleased.

alex-ju replied:
Yep, good suggestion.

vanitabarrett commented on docs/publishing-to-rubygems.md:
```diff
@@ -0,0 +1,29 @@
+  - [`lib/govuk_publishing_components/version.rb`](../lib/govuk_publishing_components/version.rb)
+  - [`Gemfile.lock`](../Gemfile.lock)
+
+7. Create a pull request and copy the changelog text for the current version.
```
"Copy the changelog text into the PR description"? Otherwise it's not clear what you're copying it to.

vanitabarrett commented on docs/publishing-to-rubygems.md:
```diff
@@ -0,0 +1,29 @@
+
+7. Create a pull request and copy the changelog text for the current version.
+
+8. Once the pull request is approved, merge to master. This action will trigger the CI to publish the new version to RubyGems.
```
And trigger dependabot PR's?

alex-ju replied:
Was thinking about adding _A [dependabot](https://github.com/dependabot) pull request will automatically be raised in frontend applications._ but it's not really part of the publishing/release process, more on the consumption/usage side.

alex-ju commented:
Updated based on suggestions (example PR after the steps) and added a paragraph to check other approved PRs before publishing to minimise the number of releases.
